vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -405,6 +409,80 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
             }
           }
 
+          // http2 pushing support
+          if (request.version() == HttpVersion.HTTP_2 && http2PushMapping != null) {
+
+            JsonArray dependencies = http2PushMapping.getJsonArray(file.substring(webRoot.length() + 1));
+            if (dependencies != null) {
+
+              for(Object dependency : dependencies) {
+                final String dep = webRoot + "/" + dependency;
+
+                HttpServerResponse response = request.response();
+
+                // get the file props
+                System.out.println("looking up props for " + dep);
+                getFileProps(context, dep, filePropsAsyncResult -> {
+                  if (filePropsAsyncResult.succeeded()) {
+
+                    // push
+                    System.out.println("pushing /" + dependency);
+
+                    writeCacheHeaders(request, filePropsAsyncResult.result());
+
+                    response.push(HttpMethod.GET, "/" + dependency, pushAsyncResult -> {
```

Comment:
It should only push object that have `nopush=false` but this wasn't available in the json